runtime/src/system_instruction_processor.rs

```diff
@@ -122,27 +122,30 @@ fn allocate_and_assign(
 
 fn create_account(
     from: &KeyedAccount,
-    to: &mut Account,
+    to: &KeyedAccount,
     to_address: &Address,
     lamports: u64,
     space: u64,
     owner: &Pubkey,
     signers: &HashSet<Pubkey>,
 ) -> Result<(), InstructionError> {
     // if it looks like the `to` account is already in use, bail
-    if to.lamports > 0 {
-        debug!(
-            "Create Account: invalid argument; account {:?} already in use",
-            to_address
-        );
-        return Err(SystemError::AccountAlreadyInUse.into());
-    }
+    {
```

Comment:
This is all single-threaded RefCells, so no problem with a lock/unlock between accesses.